DAO.sol

```diff
@@ -607,52 +610,40 @@ contract DAO is DAOInterface, Token, TokenCreation {
             sumOfProposalDeposits -= p.proposalDeposit;
         p.open = false;
     }
-	
-    function soloSplit(
+
+    function withdraw(
         uint _proposalID,
         address _newCurator
     ) noEther onlyTokenholders returns (bool _success) {
-		
-		// Did you already vote on another proposal?
+
+        unblockMe();
+
+        // Did you already vote on another proposal?
         if (blocked[msg.sender] != _proposalID && blocked[msg.sender] != 0)  {
```

Comment:
>  essentially stops the caller from splitting if they are blocked in any proposal

If that's the goal we only need: `blocked[msg.sender]  !=  0`

But the current logic is `(blocked[msg.sender] != _proposalID && blocked[msg.sender] != 0)` and that logic can be controlled one way or the other by the caller since they can essentially call `withdraw()` with any proposal ID. So for example let's say the are blocked on proposal 5. They call withdraw with proposal id = 5 and they we will not enter the if statement. If they want they can decide to go into that 'if' statement by calling withdraw() with a proposal id of 6.